GTG/gtk/browser/browser.py

```diff
@@ -420,8 +418,63 @@ def _init_accelerators(self):
                                        Gtk.AccelFlags.VISIBLE)
 
 # HELPER FUNCTIONS ##########################################################
-    def temporary_search(self, widget):
-        self.quickadd_entry.grab_focus()
+
+    def on_search_toggled(self, widget):
+        search = self.builder.get_object("search_button")
+        bar = self.builder.get_object("searchbar")
+        if bar.get_search_mode():
+            search.set_active(False)
+            bar.set_search_mode(False)
+        else:
+            search.set_active(True)
+            bar.set_search_mode(True)
+            self.search_entry.grab_focus()
+
+    def on_search(self, key, widget):
+        query = self.search_entry.get_text()
+        Log.debug("Searchign for '%s'", query)
+
+        try:
+            parsed_query = parse_search_query(query)
+
+        except InvalidQuery as e:
+            Log.warning("Invalid query '%s' : '%s'", query, e)
+            return
+
+        self.apply_filter_on_panes(SEARCH_TAG, parameters=parsed_query,
+                                   refresh=True)
```

Comment:
`refresh` down there is specified to be `True`. Can you remove it as it isn't needed?